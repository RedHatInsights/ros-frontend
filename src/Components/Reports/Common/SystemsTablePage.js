import React, { Fragment } from 'react';
import propTypes from 'prop-types';
import { Section, Column, Table } from '@redhat-cloud-services/frontend-components-pdf-generator';
import { Text } from '@react-pdf/renderer';
import styles from './styles';

const columnBuilder = ({ value, style, key }) => <Text key={key} style={style}>{value}</Text>;

const headerContent = ['Name', 'Group', 'OS', 'CPU utilization', 'Memory utilization', 'I/O utilization', 'Suggestions', 'State', 'Last reported'];

const buildSystemsHeader = () => {
    const formattedHeader = headerContent.map((item, index) => {
        let styleArr = item === 'Name' ? [styles.systemNameCell] : item === 'Last reported' ? [styles.lastReportedCell] : [styles.headerCell];
        return columnBuilder({ value: item, style: styleArr, key: `${index}-${item}` });
    });

    return formattedHeader;

};

const buildSystemsRows = (rowsData) => {
    const systemsRows =  rowsData.map((rowItem) => {
        const formattedRows = rowItem.map((rowValue, index) => {
            const nameIndex = headerContent.indexOf('Name');
            const lastReportedIndex = headerContent.indexOf('Last reported');
            let styleArr = index === nameIndex ? [styles.systemNameCell]
                : index === lastReportedIndex ? [styles.lastReportedCell] : [styles.bodyCell];
            return columnBuilder({ value: rowValue, style: styleArr, key: `${index}-${rowValue}` });
        });
        return formattedRows;
    });

    return systemsRows;
};

export const SystemsTablePage = ({ data, page }) => {

    // Table header
    const systemsHeader = buildSystemsHeader();

    // Table Rows
    const systemsRows = buildSystemsRows(data);

    return <Fragment key={page}>
        <Section>
            <Column>
                <Table
                    withHeader
                    rows={[
                        systemsHeader,
                        ... systemsRows
                    ]}
                />
            </Column>
        </Section>
    </Fragment>;

};

SystemsTablePage.propTypes = {
    data: propTypes.array,
    page: propTypes.number
};

columnBuilder.propTypes = {
    value: propTypes.string,
    style: propTypes.array,
    key: propTypes.string
};
