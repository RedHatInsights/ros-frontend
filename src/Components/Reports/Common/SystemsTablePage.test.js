import React from 'react';
import renderer from 'react-test-renderer';
import { SystemsTablePage } from './SystemsTablePage';

describe('Systems Table Page Component', () => {

    it('should matches snapshot for systems table page and renders correctly', () => {

        const tableProps = {
            data: [
                ['iqe-ros-bb519f81', 'RHEL 8.4', '90%', '80%', '0.314', '2', 'Undersized'],
                ['iqe-ros-bb519f81', 'RHEL 8.4', '90%', '80%', '0.314', '2', 'Under pressure'],
                ['iqe-ros-bb519f81', 'RHEL 8.4', 'N/A', 'N/A', 'N/A', '2', 'Waiting for data']
            ],
            page: 1
        };

        const component = renderer.create(<SystemsTablePage {...tableProps} />).toJSON();

        expect(component).toMatchSnapshot();

    });

});
