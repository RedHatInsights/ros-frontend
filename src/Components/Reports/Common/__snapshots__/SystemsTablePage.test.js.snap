// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Systems Table Page Component should matches snapshot for systems table page and renders correctly 1`] = `
<VIEW
  style={
    Object {
      "margin": "7 0",
    }
  }
>
  <TEXT
    style={
      Object {
        "color": "#c9190b",
        "fontSize": 9,
        "fontWeight": 700,
        "margin": "7 0",
      }
    }
  >
    
  </TEXT>
  <VIEW
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <VIEW
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <VIEW
        style={Object {}}
      >
        <VIEW
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": "0.5rem",
              "paddingLeft": "0.5rem",
              "paddingTop": "0.5rem",
            }
          }
        >
          <TEXT
            style={
              Array [
                Object {
                  "flex": 2.5,
                  "fontSize": 8,
                  "paddingBottom": 2,
                  "paddingRight": 2,
                  "paddingTop": 2,
                  "textAlign": "left",
                },
              ]
            }
          >
            Name
          </TEXT>
          <TEXT
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 8,
                  "textAlign": "left",
                },
              ]
            }
          >
            OS
          </TEXT>
          <TEXT
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 8,
                  "textAlign": "left",
                },
              ]
            }
          >
            CPU utilization
          </TEXT>
          <TEXT
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 8,
                  "textAlign": "left",
                },
              ]
            }
          >
            Memory utilization
          </TEXT>
          <TEXT
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 8,
                  "textAlign": "left",
                },
              ]
            }
          >
            I/O utilization
          </TEXT>
          <TEXT
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 8,
                  "textAlign": "left",
                },
              ]
            }
          >
            Suggestions
          </TEXT>
          <TEXT
            style={
              Array [
                Object {
                  "flex": 1,
                  "fontSize": 8,
                  "textAlign": "left",
                },
              ]
            }
          >
            State
          </TEXT>
          <TEXT
            style={
              Array [
                Object {
                  "flex": 1.2,
                  "fontSize": 8,
                  "paddingBottom": 2,
                  "paddingRight": 2,
                  "paddingTop": 2,
                  "textAlign": "left",
                },
              ]
            }
          >
            Last reported
          </TEXT>
        </VIEW>
        <VIEW
          style={
            Object {
              "borderBottomColor": "#8a8d90",
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "borderTopColor": "#8a8d90",
              "borderTopStyle": "solid",
              "borderTopWidth": 1,
            }
          }
        >
          <VIEW
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": "0.5rem",
                "paddingLeft": "0.5rem",
                "paddingTop": "0.5rem",
              }
            }
          >
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 2.5,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              iqe-ros-bb519f81
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              RHEL 8.4
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              90%
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              80%
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              0.314
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              2
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Undersized
            </TEXT>
          </VIEW>
          <VIEW
            style={
              Object {
                "backgroundColor": "#f0f0f0",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": "0.5rem",
                "paddingLeft": "0.5rem",
                "paddingTop": "0.5rem",
              }
            }
          >
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 2.5,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              iqe-ros-bb519f81
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              RHEL 8.4
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              90%
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              80%
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              0.314
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              2
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Under pressure
            </TEXT>
          </VIEW>
          <VIEW
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": "0.5rem",
                "paddingLeft": "0.5rem",
                "paddingTop": "0.5rem",
              }
            }
          >
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 2.5,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              iqe-ros-bb519f81
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              RHEL 8.4
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              N/A
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              N/A
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              N/A
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              2
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Waiting for data
            </TEXT>
          </VIEW>
        </VIEW>
      </VIEW>
    </VIEW>
  </VIEW>
</VIEW>
`;
